### DISCUSSION TITLE
Floor in separate isolate

### DISCUSSION THREAD

**Discussion by alectogeek** [vitusortner/floor#465] [Q&A] [answered]
Does Floor run in separate isolate?
If not, why?
I asked my colleagues, but everybody uses `moor` and they know that `moor` runs on separate isolate. Does the same `floor`?

**Comment by vitusortner** [comment, marked as answer]
Hi! As floor is based on sqflite, Android and iOS apps access the SQLite database on a native background thread. On Linux, macOS, and Windows, a separate isolate is used. You can find some further reading here https://github.com/tekartik/sqflite/blob/fc97c7973fffbf70e7a08f5d0f8b71e33082c385/sqflite/doc/usage_recommendations.md.

I'll extend the project's README.

**Reply by alectogeek** [reply]
It's clear now, thanks!
